crates/biome_deserialize/src/impls.rs

```diff
@@ -0,0 +1,571 @@
+    }
+}
+
+impl<T: Deserializable> Deserializable for Vec<T> {
+    fn deserialize(
+        value: impl DeserializableValue,
+        diagnostics: &mut Vec<DeserializationDiagnostic>,
+    ) -> Option<Self> {
+        struct Visitor<T>(PhantomData<T>);
+        impl<T: Deserializable> DeserializationVisitor for Visitor<T> {
+            type Output = Vec<T>;
+            const EXPECTED_TYPE: ExpectedType = ExpectedType::ARRAY;
+            fn visit_array(
+                self,
+                values: impl Iterator<Item = impl DeserializableValue>,
+                _range: TextRange,
+                diagnostics: &mut Vec<DeserializationDiagnostic>,
+            ) -> Option<Self::Output> {
+                Some(
+                    values
+                        .filter_map(|value| Deserializable::deserialize(value, diagnostics))
+                        .collect(),
+                )
+            }
+        }
+        value.deserialize(Visitor(PhantomData), diagnostics)
+    }
+}
+
+impl<T: Deserializable + Eq + Hash, S: BuildHasher + Default> Deserializable for HashSet<T, S> {
+    fn deserialize(
+        value: impl DeserializableValue,
+        diagnostics: &mut Vec<DeserializationDiagnostic>,
+    ) -> Option<Self> {
+        struct Visitor<T, S>(PhantomData<(T, S)>);
+        impl<T: Deserializable + Eq + Hash, S: BuildHasher + Default> DeserializationVisitor
+            for Visitor<T, S>
+        {
+            type Output = HashSet<T, S>;
+            const EXPECTED_TYPE: ExpectedType = ExpectedType::ARRAY;
+            fn visit_array(
+                self,
+                values: impl Iterator<Item = impl DeserializableValue>,
+                _range: TextRange,
+                diagnostics: &mut Vec<DeserializationDiagnostic>,
+            ) -> Option<Self::Output> {
+                Some(
+                    values
+                        .filter_map(|value| Deserializable::deserialize(value, diagnostics))
+                        .collect(),
+                )
+            }
+        }
+        value.deserialize(Visitor(PhantomData), diagnostics)
+    }
+}
+
+impl<T: Hash + Eq + Deserializable> Deserializable for IndexSet<T> {
+    fn deserialize(
+        value: impl DeserializableValue,
+        diagnostics: &mut Vec<DeserializationDiagnostic>,
+    ) -> Option<Self> {
+        struct Visitor<T>(PhantomData<T>);
+        impl<T: Hash + Eq + Deserializable> DeserializationVisitor for Visitor<T> {
+            type Output = IndexSet<T>;
+            const EXPECTED_TYPE: ExpectedType = ExpectedType::ARRAY;
+            fn visit_array(
+                self,
+                values: impl Iterator<Item = impl DeserializableValue>,
+                _range: TextRange,
+                diagnostics: &mut Vec<DeserializationDiagnostic>,
+            ) -> Option<Self::Output> {
+                Some(
+                    values
+                        .filter_map(|value| Deserializable::deserialize(value, diagnostics))
```

Comment:
This `filter_map` removes items that contain syntax errors. This means that during the deserialization, the original array and the deserialized array can be different in length. Is this intended? It seems to be very unsafe.

Reply:
Yes, it is part of the fault-tolerant philosophy. I am not sure to see the problem with that. We also deserialize an object even if one of its property has a syntax error.

Reply:
Cool. Just making sure that that's what we want